bundles/tools.vitruv.applications.umljava/src/tools/vitruv/applications/umljava/uml2java/UmlToJavaClassifier.reactions

```diff
@@ -106,6 +106,36 @@ routine createJavaCompilationUnit(uml::Classifier umlClassifier, java::ConcreteC
     }
 }
 
+reaction UmlClassifierInserted {
+    after element uml::Classifier inserted in uml::Package[packagedElement]
+    call moveJavaClassifier(newValue, affectedEObject)
+}
+
+routine moveJavaClassifier(uml::Classifier umlClassifier, uml::Package umlPackage) {
+	match {
+		val javaClassifier = retrieve java::ConcreteClassifier corresponding to umlClassifier
+		val javaPackage = retrieve optional java::Package corresponding to umlPackage
+		val javaCompilationUnit = retrieve java::CompilationUnit corresponding to umlClassifier 
+			with !javaPackage.isPresent || !javaPackage.get.compilationUnits.contains(javaCompilationUnit)
+	}
+	action {
+		update javaCompilationUnit {
+			javaCompilationUnit.namespaces.clear
+			javaPackage.ifPresent [
+				javaCompilationUnit.namespaces += javaPackageAsStringList
+			]
+			//TODO: updating the compilation unit's name should not be needed 
```

Comment:
I checked how this is handled in other routines, and it seems to be the standard. For example here:
https://github.com/vitruv-tools/Vitruv-Applications-ComponentBasedSystems/blob/97ddb33878267e365b8f704eb93792bf673d7f70/bundles/tools.vitruv.applications.umljava/src/tools/vitruv/applications/umljava/uml2java/UmlToJavaClassifier.reactions#L129
Thus I think we can remove the todo-comment.

Reply:
I think the referenced example is chose a bit unluckily as it's a rename operation and rename operations obviously need to update the name.
I searched the PCM -> Java and the other UML -> Java reactions for an actual move reaction as reference but did not find any so I don't know if there even is any _standard behaviour_ for a ```move``` operation currently.

Reply:
Good point. Then let us keep the comment for now as we can always decide to remove it.